### ISSUE TITLE
Nemo GTK theming

### ISSUE THREAD

**Issue by hrqmonteiro** [linuxmint/nemo#2563]
```
 * Nemo version - 4.6.5.
 * Desktop
 * Distribution - Void Linux
 * NVIDIA 920m gpu
 * 64 bit
 ```

**Issue**

Hi, i am trying to set my GTK theme, and i wanted to change some things particularly on Nemo, but i am not able to use the GTK Inspector on it to find the classes.
And i saw some posts on internet refering to a gtk-fallback.css file for Nemo, which, apparetly contained all the classes for it. But all the links are broken.

Can anyone show me how would i get these classes?

**Comment by mtwebster** [comment]
There was an unfortunate mix up a while back in keyboard shortcuts, and now both ctrl-shift-d and -i are used in nemo.

You can launch the inspector using a GTK_DEBUG flag - 
```
nemo --quit
GTK_DEBUG=interactive nemo
```

I believe the classes in both of these files can be overridden by the theme:
https://github.com/linuxmint/nemo/blob/master/gresources/nemo-style-fallback.css
https://github.com/linuxmint/nemo/blob/master/gresources/nemo-style-fallback-mandatory.css
The second one - 'mandatory' - is loaded by Nemo only if the current theme doesn't provide *any* nemo support - they're for some things that would be essentially broken or unusable without minimum styling.

There is a third:
https://github.com/linuxmint/nemo/blob/master/gresources/nemo-style-application.css
The contents in this file *cannot* be overridden (they're set to a higher priority than the theme). They deal with desktop elements to ensure that the text on the desktop will always be readable regardless of background color.

If you're interested, here is the section of code that applies all theme-related things:
https://github.com/linuxmint/nemo/blob/master/src/nemo-application.c#L102-L309

**Comment by hrqmonteiro** [comment]
> 
> 
> There was an unfortunate mix up a while back in keyboard shortcuts, and now both ctrl-shift-d and -i are used in nemo.
> 
> You can launch the inspector using a GTK_DEBUG flag -
> 
> ```
> nemo --quit
> GTK_DEBUG=interactive nemo
> ```
> 
> I believe the classes in both of these files can be overridden by the theme:
> https://github.com/linuxmint/nemo/blob/master/gresources/nemo-style-fallback.css
> https://github.com/linuxmint/nemo/blob/master/gresources/nemo-style-fallback-mandatory.css
> The second one - 'mandatory' - is loaded by Nemo only if the current theme doesn't provide _any_ nemo support - they're for some things that would be essentially broken or unusable without minimum styling.
> 
> There is a third:
> https://github.com/linuxmint/nemo/blob/master/gresources/nemo-style-application.css
> The contents in this file _cannot_ be overridden (they're set to a higher priority than the theme). They deal with desktop elements to ensure that the text on the desktop will always be readable regardless of background color.
> 
> If you're interested, here is the section of code that applies all theme-related things:
> https://github.com/linuxmint/nemo/blob/master/src/nemo-application.c#L102-L309

Hello, i forgot to mentioning but i was already trying to launch the inspector through the terminal command `GTK_DEBUG=interactive nemo` but it doesn't work either.

I can launch any other gtk app through the GTK_DEBUG on terminal but Nemo doesn't.

**Comment by mtwebster** [comment]
That should work, but you have to make sure nemo isn't already running - `nemo --quit`.

Simply closing all nemo windows isn't sufficient (the process terminates 30 seconds after the last window is closed).

**Comment by ItzSwirlz** [comment]
I'll have to see. Should start thinking about updating my themes... Tbh, my GTK theme applying works with Nemo-but themes need to add Nemo theme support although it can mostly take info from the current GTK themes.

But I'm sure Nautilus needs theming too, this isn't a big issue.

**Comment by ItzSwirlz** [comment]
Looking at linuxmint/mint-themes, does Mint-X themes even have support for Nemo? If so where do the css files go?

**Comment by mtwebster** [comment]
There's a cinnamon-applications.css file.  For Mint-Y it's part of the main css definitions. Mint-Y is generated from pseudo-code I think, Mint-X is not.

**Comment by ItzSwirlz** [comment]
Yep-here it is. https://github.com/linuxmint/mint-themes/blob/master/src/Mint-X/theme/Mint-X/gtk-3.0/apps/cinnamon-applications.css

**Comment by hrqmonteiro** [comment]
> There was an unfortunate mix up a while back in keyboard shortcuts, and now both ctrl-shift-d and -i are used in nemo.
> 
> You can launch the inspector using a GTK_DEBUG flag -
> 
> ```
> nemo --quit
> GTK_DEBUG=interactive nemo
> ```
> 
> I believe the classes in both of these files can be overridden by the theme:
> https://github.com/linuxmint/nemo/blob/master/gresources/nemo-style-fallback.css
> https://github.com/linuxmint/nemo/blob/master/gresources/nemo-style-fallback-mandatory.css
> The second one - 'mandatory' - is loaded by Nemo only if the current theme doesn't provide _any_ nemo support - they're for some things that would be essentially broken or unusable without minimum styling.
> 
> There is a third:
> https://github.com/linuxmint/nemo/blob/master/gresources/nemo-style-application.css
> The contents in this file _cannot_ be overridden (they're set to a higher priority than the theme). They deal with desktop elements to ensure that the text on the desktop will always be readable regardless of background color.
> 
> If you're interested, here is the section of code that applies all theme-related things:
> https://github.com/linuxmint/nemo/blob/master/src/nemo-application.c#L102-L309

These .css files are useful for what i want to do. But let me get some things straight. Since i don't use cinnamon or nemo for more than two years, and back them, i wasn't into css or linux ricing/customization:

- What is the difference between 'desktop' and 'paned'?
- I only want to change the background color of the sidebar, and the background color of the area when the folders are placed, is it possible you can point me which classes are these ones? Since i can't open the inspector in any way here.

Thank you in advance.

**Comment by mtwebster** [comment]
I don't think the sidebar is a nemo-specific style class.  Desktop has to do with the desktop icons (This is a separate program - 'nemo-desktop'. Paned probably refers to the split view mode (View menu->Extra pane).

I can't think of any reason you can't get the inspector to show. Are you *certain* nemo isn't running when you try `GTK_DEBUG=interactive nemo`? If this command immediately returns, then it was already running (and the inspector won't show).

**Comment by hrqmonteiro** [comment]
> I don't think the sidebar is a nemo-specific style class. Desktop has to do with the desktop icons (This is a separate program - 'nemo-desktop'. Paned probably refers to the split view mode (View menu->Extra pane).
> 
> I can't think of any reason you can't get the inspector to show. Are you _certain_ nemo isn't running when you try `GTK_DEBUG=interactive nemo`? If this command immediately returns, then it was already running (and the inspector won't show).

I was able to open the debug now, i don't know why but not it is working, after a reboot. I was using it for a while and i was able to find a class i was searching, i was able to change the background using:

```
.nemo-window .view: {
   background-color: ;
}
```

But the second thing i want to change is really the sidebar, that you said that is not nemo-specific. Well, if i can get some help i appreciate, if is not possible, is okay. But on the inspector, it returns me this, when i use the selector:

![2020-11-19_13-32](https://user-images.githubusercontent.com/17055027/99694981-c763b600-2a6b-11eb-98bf-dd276fe5ade1.png)

And i am refering to only the color of this sidebar:

![2020-11-19_13-33](https://user-images.githubusercontent.com/17055027/99695044-da768600-2a6b-11eb-8327-03c8cf84ebea.png)

So, you think i probably won't be able to achieve it editing just the nemo css classes? So i have to edit in the gtk general file in my theme, right?

**Comment by hrqmonteiro** [comment]
Forgot to mention, but i tried to edit the `places-treeview` class and really wasn't able to achieve nothing...

**Comment by mtwebster** [comment]
Honestly my theme knowledge is very limited. You can be specific when overriding certain widgets by specifying parent nodes as well in the definition - this way you're not overriding every program that uses this specific widget. But really you would probably find the most help by downloading other themes and seeing what they've done: maybe here? https://www.gnome-look.org/browse/cat/135/order/latest/

**Comment by JosephMcc** [comment]
I believe you can change the background of the sidebar using .nemo-window .sidebar {} and setting a background-color. You can see how it is set here in Mint-Y: https://github.com/linuxmint/mint-themes/blob/master/src/Mint-Y/gtk-3.0/sass/_applications.scss#L272

**Comment by hrqmonteiro** [comment]
> I believe you can change the background of the sidebar using .nemo-window .sidebar {} and setting a background-color. You can see how it is set here in Mint-Y: https://github.com/linuxmint/mint-themes/blob/master/src/Mint-Y/gtk-3.0/sass/_applications.scss#L272

No, didn't work here, unfortunately..

**Comment by theotrst** [comment]
hi hrqmonteiro,
could I ask you how you prepared you `custom.css ` file? Did you create a CSS in the editor of the GTK_DEBUG? I am not sure in changing original css files, so I would prefer doing a separate CSS that overwrites certain parameters. 

Actually still I did not locate the css which includes

```
.nemo-window .view: {
   background-color: ;
}
```

I found "nemo-style-application.css" and there I found only

```
.nemo-window.nemo-desktop-window notebook,
.nemo-window.nemo-desktop-window paned {
	background-color: transparent;
}
```

That is strange because I have grey colour. I see that colour in the "Objects" as CSS-nodes as a general window color rgb(53,53,53). I am a bit lost.
